LuatixHQ/opencti
# [backend/frontend] Add a new capability to restrict the access to users list (#3151)

### Proposed changes

* Add a new capability "Assign users" under "Knowledge update"
* Use aggregations for creators and assignee filters : filters will only return actual creators and assignees for the entity type searched.
* Retrict creators / assignees / members APIs : creators and members endpoints will return the full list only for "knowledge update" capability and assignees is restricted with the newly added "knowledge update assign" capability

#### To test the issue : 
- log with user that doesn't have bypass capability
- In a list of entities (for example reports), in the autocomplete filters "assignee" and "technical creator" you should have only actual assignees and creators of the searched entities (instead of all the users of the platform)
- When you select an entity (like a report), and try to edit it, you should not be able to see "assign to" field if you don't have the new capability "Assign users"

### Related issues

* https://github.com/OpenCTI-Platform/opencti/issues/3151

### Checklist







### Further comments
